gulpfile.js/deploy.js

```diff
@@ -0,0 +1,94 @@
+const { series } = require('gulp')
+const GulpError = require('plugin-error')
+const { writeFileSync } = require('fs')
+const { dryRun, getGitHash, spawner, updateRef, lastVersion, nextVersion, codeVersions, tagRef, getGitName,
+    onBuildServer
+} = require('./utils')
+const { postDeployTests } = require("./tests");
+
+const VALID_ENVIRONMENT_ARGS = ['develop', 'production']
+const DEFAULT_ENVIRONMENT = 'develop'
+const VALID_SEMVER_CHANGE_ARG = ['patch', 'minor', 'major']
+const DEFAULT_SEMVER_CHANGE = 'patch'
+
+const POINTER_BRANCH_LOOKUP = {
+    develop: 'develop',
+    production: 'main',
+}
+
+// Provider specific and repo specific steps
+const apply = async (environment) => {
+    if (dryRun())
+        console.log(`\x1b[1m\x1b[35mThis is a dry run\x1b[0m`)
+    console.log(`\x1b[1m\x1b[33mApplying kubernetes resources\x1b[0m`)
+
+    // Authentication
+    let kubectlArgs
+    if (onBuildServer()) {
+        if (!process.env.KUBECONFIG_CERT_AUTH_DATA || !process.env.KUBECONFIG_SERVER || !process.env.KUBECONFIG_USER_TOKEN) {
+            throw new GulpError('apply', new Error('Error: Three environment variables are required for build server deployments from a valid kubeconfig.yaml file:\n' +
+                'KUBECONFIG_CERT_AUTH_DATA (from clusters:- cluster: certificate-authority-data:),\n' +
+                'KUBECONFIG_SERVER (from clusters:- cluster: server:), and \n' +
+                'KUBECONFIG_USER_TOKEN (from users: user: token:)'))
+        }
+        kubectlArgs = `--server ${process.env.KUBECONFIG_SERVER} --token ${process.env.KUBECONFIG_USER_TOKEN} --client-key 'ca_file.cert' --insecure-skip-tls-verify`
+        writeFileSync(`${process.env.HOME}/ca_file.cert`, process.env.KUBECONFIG_CERT_AUTH_DATA)
+    } else {
+        if (!process.env.KUBECONFIG) {
+            throw new GulpError('apply', new Error('Error: A KUBECONFIG environment variable must be set that points to a vailid kubeconfig yaml file.'))
+        }
+        kubectlArgs = `--kubeconfig ${process.env.KUBECONFIG}`
+    }
+
+    // create a namespace and deploy the containers to the cluster.
+    try {
+        await spawner(`kubectl create ns ${environment} ${kubectlArgs}`, true)
+    } finally {
+        await spawner(`kubectl apply -f ./k8s -n ${environment} ${kubectlArgs}`)
+    }
```

Comment:
Is the try needed here at all? The spawner is already setting `noThrow` to true.

```suggestion
    await spawner(`kubectl create ns ${environment} ${kubectlArgs}`, true)
    await spawner(`kubectl apply -f ./k8s -n ${environment} ${kubectlArgs}`)
```

Reply:
the try is not needed, it has been removed.